services/email/email.js

```diff
@@ -1,38 +1,28 @@
 const broker = require('../init')
-
-const mailgun = require('mailgun-js')({
-  apiKey: process.env.API_KEY || 'secret-key-here',
-  domain: process.env.DOMAIN || 'www.gmail.com',
-  host: process.env.HOST || 'api.mailgun.net'
-})
-
-const sender = process.env.SENDER_EMAIL || 'username username@gmail.com'
 const auth = process.env.AUTH || 'secret-key-here'
+const mailGun = require('./methods/mailgun')
+const senGrid = require('./methods/sengrid')
 
 broker.createService({
-  settings: {
-    port: process.env.PORT || '3000'
-  },
   name: 'email',
   actions: {
-    async send (ctx) {
+    send (ctx) {
+      let response = ''
       if (ctx.params.auth !== auth) {
         ctx.meta.$statusCode = 401
         return { error: 'Error: auth key invalid' }
       }
-      const data = {
-        from: sender,
-        to: ctx.params.to,
-        subject: ctx.params.subject,
-        text: ctx.params.text,
-        html: ctx.params.html
-      }
+
       try {
-        const res = await mailgun.messages().send(data)
-        return (res)
+        if (Math.floor(Math.random() * 2) % 2) {
```

Comment:
Please fix the logic in the `if` statement. Dividing the load between mailgun and sendgrid can be done differently .
You could split the list in half and assign them to Sendgrid and Mailgun

Reply:
fixed in https://github.com/ashikka/email-service/commit/6f3ec6d649946dfe056e8ba05c95d533615486d2